go/vt/tabletserver/schema_info.go

```diff
@@ -229,11 +230,28 @@ func (si *SchemaInfo) Close() {
 	si.tables = nil
 }
 
+// IsClosed returns true if the SchemaInfo is closed.
+func (si *SchemaInfo) IsClosed() bool {
+	si.mu.Lock()
+	defer si.mu.Unlock()
+	return si.tables == nil
+}
+
 // Reload reloads the schema info from the db. Any tables that have changed
-// since the last load are updated.
-func (si *SchemaInfo) Reload() {
+// since the last load are updated. This is a no-op if the SchemaInfo is closed.
+func (si *SchemaInfo) Reload(ctx context.Context) error {
 	defer logError(si.queryServiceStats)
-	ctx := context.Background()
+
+	if si.IsClosed() {
```

Comment:
This is racy because si can get closed immediately after the check. The better way will be to use the startRequest pattern at the tabletserver level.